src/renderer/base/renderer.cpp

```diff
@@ -677,8 +677,7 @@ void Renderer::_PaintBufferOutput(_In_ IRenderEngine* const pEngine)
 
     for (const auto& dirtyRect : dirtyAreas)
     {
-        // Shortcut: don't bother redrawing if the width is 0.
-        if (dirtyRect.left == dirtyRect.right)
+        if (!dirtyRect)

```

Comment:
BTW This is a bug. Some engines return dirty rects where `top == bottom`, and I don't think any engine returns `left == right` actually. This causes an invalid `Viewport` to be created a few lines below. In either case though, this now checks both dimensions.

Reply:
wuh oh. good catch then.